databases/week3/homework.sql

```diff
@@ -0,0 +1,162 @@
+	insert into task ( title, description, created, updated, due_date, status_id) values ( 'Wash car', 'Dont forget', '2017-10-25 07:54:16', '2017-10-15 13:05:09', null, 2);
+
+	-- Change the title of a task
+	update task
+	set title = 'Plan meeting with Priyanka'
+	where id = 3;
+
+	-- Change a task due date
+	update task
+	set due_date = '2020-02-08 14:31:02'
+	where id = 1;
+    
+--     Change a task status 		+		 Mark a task as complete
+
+	update task
+	set status_id = '3'
+	where id = 1;
+    
+  
+--     Delete a task
+	delete from task
+	where id = 21;
+    
+    
+    
+--  ++++++++++++++++++++++++++++++++		Part 2: School database		++++++++++++++++++++++++++++++++ 
+
+
+-- Create a new database containing the following tables:
+create database hyf_homework_DB2;
+use hyf_homework_DB2;
+
+SET NAMES utf8mb4;
+
+
+-- Class: with the columns: id, name, begins (date), ends (date)
+
+CREATE TABLE `Class` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `name` varchar(255) NOT NULL,
+  `begins` DATETIME NULL DEFAULT NULL,
+  `ends` DATETIME NULL DEFAULT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+select * from Class;
+
+-- Class
+insert into Class (id, name, begins, ends) values (1, 'Aarika Ellingworth', '2017-10-15 13:05:09', '2017-10-25 06:54:16');
+insert into Class (id, name, begins, ends) values (2, 'Pren Goldsworthy', '2017-10-20 04:14:31', '2017-10-08 06:14:31');
+insert into Class (id, name, begins, ends) values (3, 'Pablo Kisbee', '2017-12-25 16:00:46', '2017-12-24 15:01:16');
+insert into Class (id, name, begins, ends) values (4, 'Rodie Duncan', '2017-11-30 16:00:46', '2017-10-28 10:30:40');
+
+
+
+-- Student: with the columns: id, name, email, phone, class_id (foreign key)
+
+CREATE TABLE `Student` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `name` varchar(255) NOT NULL,
+  `email` varchar(255) NOT NULL,
+  `phone` varchar(255) NOT NULL,
+  `class_id` int(10) unsigned NOT NULL,  
+  CONSTRAINT `fk_class_id` FOREIGN KEY (`class_id`) REFERENCES `class` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+  
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+select * from Student;
+
+-- Student
+insert into Student (id, name, email, phone, class_id) values (1, 'Aarika Ellingworth', 'aellingworth0@harvard.edu', '483-396-8795', 1);
+insert into Student (id, name, email, phone, class_id) values (2, 'Pren Goldsworthy', 'pgoldsworthy1@spotify.com', '635-572-8467', 2);
+insert into Student (id, name, email, phone, class_id) values (3, 'Pablo Kisbee', 'pkisbee2@lulu.com', '790-962-8683', 3);
+insert into Student (id, name, email, phone, class_id) values (4, 'Rodie Duncan', 'rduncan3@quantcast.com', '646-743-6191', 4);
+
+
+
+-- to ask from mantors
+
+-- what these 2 lines below do
+--   CONSTRAINT `fk_status` FOREIGN KEY (`status_id`) REFERENCES `status` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
This is the definition of the relationship with another table. the `status_id `gets realted to the `id `from the `status` table and `on delete cascade` means that if the row in the parent table gets deleted the relative rows in the child table will automatically get deleted too by the database.

Reply:
Thankyou 😄